django_colortag/widgets.py

```diff
@@ -145,7 +145,10 @@ def __init__(self, attrs=None, choices=None):
 
     def set_subwidgets(self, choices):
         self.widgets = [ColortagIncludeExcludeWidget(tag=c) for c in choices]
-        self.widgets_names = ['_%s' % i for i in range(len(self.widgets))]
+        self.widgets_names = ['_%s' % choices[i].slug for i in range(len(self.widgets))]
+
+    def get_context(self, name, value, attrs):
```

Comment:
This indeed does seem to set the URL as tags_[slug]=[I/E][tagID], as it would be nice to work, but unfortunately that seems to break the filter (so the filters aren't applied), which does meant that the previously selected tags actually aren't highlighted/selected anymore in the filter panel. Only after commenting on the other converersation, I realized that the extra 1 in the filter name (e.g. 'tags_1_test') is due to the ColortagIEAndOrWidget consisting of two separate widgets: the first one is the checkbox whether to use "Match any (OR") or "Match all (AND)", the second one is this colortagIEMutliWidget, So in the URL, tags_0 refers to the checkbox, tags_1 leads to this colortagIEMultiWidget, which then would have the slugs after it. So it indeed does seem to need the extra 1 in the middle. So if this get_context method is a dirty hack to remove that 1, this code probably would work if this method is removed. It's probably too big of a hassle to figure out how to get the filtering to work without that extra 1, so it's probably better to just include it.

Reply:
Now the tag number is replaced with a slug and the filtering works.